File: src-tauri/src/ram.rs
use serde::{Serialize, Deserialize};
use sysinfo::{System};

#[derive(Serialize, Deserialize)]
pub struct Ram {
    memory: Memory,
    swap: Swap
}

#[derive(Serialize, Deserialize, Debug)]
pub struct Memory {
    total: u64,
    used: u64,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct Swap {
    total: u64,
    used: u64
}

#[tauri::command]
pub fn get_ram() -> Ram {
    let mut sys: System = System::new();
    sys.refresh_memory();

    return Ram {
        memory: Memory {
            total: sys.total_memory(),
            used: sys.used_memory()
        },
        swap: Swap {
            total: sys.total_swap(),
            used: sys.used_swap()
        }
    };
}

File: src/lib/backend/ram.ts
import { invoke } from '@tauri-apps/api/tauri'

export type Ram = {
    memory: Memory
    swap: Swap
}

export type Memory = {
    total: number
    used: number
}

export type Swap = {
    total: number
    used: number
}

export async function getCurrentRamState(): Promise<Ram | string> {
    try {
        const response: Ram = await invoke<Ram>('get_ram')

        return response
    } catch (error) {
        return 'Error'
    }
}
